source/MissionAction.cpp

```diff
@@ -356,32 +420,126 @@ bool MissionAction::CanBeDone(const PlayerInfo &player, const shared_ptr<Ship> &
-				if(!ship->IsDestroyed())
-				{
-					available += ship->Cargo().Get(it.first);
-					available += ship->OutfitCount(it.first);
-				}
+			// The following if statements swap location checks to true if certain keywords have
+			// been omitted, e.g, not specifying installed or cargo means check both.
+			if(!checkFlag && !checkEscorts) {
+				checkEscorts = true;
+				if(!checkAbsent && !checkParked)
+					checkFlag = true;
+			}
+			if(!checkCargo && !checkInstalled)
+			{
+				checkCargo = true;
+				checkInstalled = true;
+			}
+			if(!checkPresent && !checkAbsent)
+			{
+				checkPresent = true;
+				checkAbsent = true;
+			}
+			if(!checkParked && !checkUnparked)
+			{
+				checkParked = true;
+				checkUnparked = true;
+			}
+		}
+		else if(!count)
+		{
+			// Requiring the player to have 0 of this outfit and not specifying where to look
+			// means all ships and all cargo holds must be checked, even if the ship is disabled, 
+			// parked, or out-of-system.
+			checkFlag = true;
+			checkEscorts = true;
+			checkInstalled = true;
+			checkCargo = true;
+			checkPresent = true;
+			checkAbsent = true;
+			checkParked = true;
+			checkUnparked = true;
 		}
 		else
 		{
-			// Required outfits must be present on able ships in the
-			// player's location (or the respective cargo hold).
-			available += flagship ? flagship->OutfitCount(it.first) : 0;
-			available += boardingShip ? flagship->Cargo().Get(it.first)
+			// If no check locations were specified and the required amount is greater than zero, 
+			// check only the flagship.
+			checkFlag = true;
+			checkCargo = true;
+			checkInstalled = true;
+		}
+		
+		// Now that the check booleans have been handled, check the flagship
+		// if necessary.
+		if(checkFlag)
+		{
+			if(checkCargo)
+				available += boardingShip ? flagship->Cargo().Get(it.first)
 					: CountInCargo(it.first, player);
+			if(checkInstalled)
+				available += flagship ? flagship->OutfitCount(it.first) : 0;
+			
+			// If the required amount is 0 and an outfit has already been found,
+			// then return false.
+			if(!count && available > 0)
+				return false;
 		}
 		
-		if(available < it.second)
-			return false;
 		
-		// If the required count is 0, the player must not have any of the outfit.
-		if(checkAll && available)
+		// If the escorts need to be checked, then iterate through all the player's ships
+		if(checkEscorts)
+		{
+			checkFlag = true;
+			for(const auto &ship : player.Ships())
+			{
+				// Ignore destroyed ships.
+				if(ship->IsDestroyed())
+					continue;
+				// The first ship in list is the flagship, so skip over it.
+				if(checkFlag)
+				{
+					checkFlag = false;
+					continue;
+				}
+
+				if(checkCargo)
+				{
+					if(checkPresent && ship->GetSystem() == player.GetSystem())
+					{
+						// If the player is landed and not parked then check the player's pooled cargo
+						if(player.GetPlanet() && !ship->IsParked())
+							available += player.Cargo().Get(it.first);
```

Comment:
This code is executed on a per-ship basis, meaning that being landed will erroneously result in the player's count of this outfit in cargo being multiplied by the number of escorts present.

Reply:
The latest commit should have fixed this by only checking the player's pooled cargo once before entering this for loop. Also fixed it by not counting the pooled cargo a second time if it was already counted for the flagship.